File: pset2/vigenere/vigenere.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>

int shift(char c);

int main(int argc, string argv[])
{
    // Check to make sure only 1 argument is entered
    if (argc == 2)
    {
        for (int i = 0; i < strlen(argv[1]); i++)
        {
            //Check to makesure the argument is a word
            if (isalpha(argv[1][i]) == 0)
            {
                printf("Usage: ./vigenere keyword\n");
                return 1;
            }
        }
    }
    else
    {
        printf("Usage: ./vigenere keyword\n");
        return 1;
    }
    // Request plain text from the user
    string plain_text = get_string("plaintext: ");
    printf("ciphertext: ");
    // set counter for key
    int COUNT = 0;
    // convert the plain text to cipher text
    for (int i = 0; i < strlen(plain_text); i++)
    {
        // Set the key
        int key = shift(argv[1][COUNT % strlen(argv[1])]);

        // convert lower case letters & print
        if (islower(plain_text[i]))
        {
            printf("%c", ((plain_text[i] - 97 + (key % 26)) % 26) + 97);
            COUNT++;
        }
        // convert upper case letters & print
        else if (isupper(plain_text[i]))
        {
            printf("%c", ((plain_text[i] - 65 + (key % 26)) % 26) + 65);
            COUNT++;
        }
        // print punctuation & spaces as is
        else
        {
            printf("%c", plain_text[i]);
        }
    }
    printf("\n");
    return 0;
}


int shift(char c)
{
    c = tolower(c);
    // Get the numeric value with a starting at 0
    int value = c - 97;
    return value;
}

File: pset6/vigenere/vigenere.py
# Uses the Vigenere Cipher to encode messages

import sys
from cs50 import get_string


# Check the number of arguments
if len(sys.argv) == 2:
    # Check that the argument is a number
    if sys.argv[1].isalpha():
        # store that argument as integer
        k = sys.argv[1].lower()
    # Error message if arguments used incorrectly
    else:
        sys.exit("Usage: python vigenere.py k")
else:
    sys.exit("Usage: python vigenere.py k")


# Get input for the text to encode
plainText = get_string("plaintext: ")

# Set up the space for the cipher text (don't end the line)
print("ciphertext: ", end='')

# Initialize a counter for k
count = 0

# Iterate through the plaintext
for c in plainText:

    key = ord(k[count % len(k)]) - 97
    # Convert lowercase letters
    if c.islower():
        print(chr((ord(c) - 97 + key) % 26 + 97), end='')
        count += 1
    # Convert uppercase letters
    elif c.isupper():
        print(chr((ord(c) - 65 + key) % 26 + 65), end='')
        count += 1
    # Print spaces and puncuation as is
    else:
        print(c, end='')

# Print a new line to be pretty
print()
